webrtc.html

```diff
@@ -10528,83 +10534,67 @@ <h2 id="sec.idp-error-handling">IdP Error Handling</h2>
         mechanism they used to establish that they were going to communicate in
         the first place.</p>
         <pre class="example highlight">
-var signalingChannel = new SignalingChannel();
-var configuration = { iceServers: [{ urls: "stuns:stun.example.org" }] };
-var pc;
-
-// call start() to initiate
-function start() {
-    pc = new RTCPeerConnection(configuration);
-
-    // send any ice candidates to the other peer
-    pc.onicecandidate = function (evt) {
-        signalingChannel.send(JSON.stringify({ candidate: evt.candidate }));
-    };
-
-    // let the "negotiationneeded" event trigger offer generation
-    pc.onnegotiationneeded = function () {
-        pc.createOffer().then(function (offer) {
-            return pc.setLocalDescription(offer);
-        })
-        .then(function () {
-            // send the offer to the other peer
-            signalingChannel.send(JSON.stringify({ desc: pc.localDescription }));
-        })
-        .catch(logError);
-    };
+          const signaling = new SignalingChannel(); // handles JSON.stringify/parse
+          const constraints = { audio: true, video: true };
+          const configuration = { iceServers: [{ urls: "stuns:stun.example.org" }] };
+          const pc = new RTCPeerConnection(configuration);
 
-    // once remote track arrives, show it in the remote video element
-    pc.ontrack = function (evt) {
-        // don't set srcObject again if it is already set.
-        if (!remoteView.srcObject)
-          remoteView.srcObject = evt.streams[0];
-    };
+          // send any ice candidates to the other peer
+          pc.onicecandidate = {candidate} =&gt; signaling.send({ candidate });
```

Comment:
We might want to use parenthesis around the `{candidate}`

Reply:
Yes, this is a syntax error otherwise, due to Javascript parsing limitation with destructuring.